integration-tests/load/ccip/ccip_test.go

```diff
@@ -220,34 +226,51 @@ func TestCCIPLoad_RPS(t *testing.T) {
 		}
 	}
 
-	evmSourceKeys := make(map[uint64]*bind.TransactOpts)
+	evmSourceKeys := make(map[uint64]map[uint64]*bind.TransactOpts)
 	solSourceKeys := make(map[uint64]*solana.PrivateKey)
 	var mu sync.Mutex
+
 	for ind, cs := range destinationChains {
-		otherChains := env.BlockChains.ListChainSelectors(cldf_chain.WithChainSelectorsExclusion([]uint64{cs}))
-		for _, src := range otherChains {
+		srcChains := laneConfig.GetSourceChainsForDestination(cs)
+
+		// Initialize the map for this destination
+		evmSourceKeys[cs] = make(map[uint64]*bind.TransactOpts)
+
+		for _, src := range srcChains {
 			selFamily, err := selectors.GetSelectorFamily(src)
 			if err != nil {
 				lggr.Errorw("Failed to get selector family", "chainSelector", src, "error", err)
+				continue
 			}
 			mu.Lock()
 			switch selFamily {
 			case selectors.FamilyEVM:
-				evmSourceKeys[src] = evmSenders[src][ind]
-
+				// Check if we have enough senders for this source chain
+				if len(evmSenders[src]) <= ind {
+					lggr.Errorw("Not enough EVM senders for source chain",
+						"sourceChain", src,
+						"destinationChain", cs,
+						"requiredIndex", ind,
+						"availableSenders", len(evmSenders[src]))
+					mu.Unlock()
+					continue
+				}
+				evmSourceKeys[cs][src] = evmSenders[src][ind]
```

Comment:
Let's reverse this block, 
```suggestion
				if len(evmSenders[src]) > ind { 
				 evmSourceKeys[cs][src] = evmSenders[src][ind]
				}```